### ISSUE TITLE
[BUG] the createTabs method creates twice the number of tabs

### ISSUE THREAD

**Issue by Pandishpan** [henrychavez/nativescript-bottom-navigation#65]
**Which platform(s) does your issue occur on?**
- [*] iOS
- [*] Android

**What type of device?**
- [*] Emulator
- [*] Device

√ Component nativescript has 5.4.2 version and is up to date.
√ Component tns-core-modules has 5.4.3 version and is up to date.
√ Component tns-android has 5.4.0 version and is up to date.

When I'm using the createTabs method to create tabs programmatically it creates twice the number of tabs. Although I'm calling the createTabs method only once.

  ```
onBottomNavigationLoaded(args) {
	if (!this.bottomMenu) {
		this.bottomMenu = args.object;
		const tabs = [
			new BottomNavigationTab("First", "ic_home"),
			new BottomNavigationTab("Second", "ic_view_list")
		];
		this.bottomMenu.createTabs(tabs);
	}
} 
```

![bug_plm](https://user-images.githubusercontent.com/31397153/61363537-19933680-a87c-11e9-9cc5-bcdcedc19c4d.jpg)

**Comment by Pandishpan** [comment]
It looks like the tabs are added first by calling the createTabs method and then it's called one more time by the tabsProperty.setNative after you set the tabs property.


```
 createTabs(tabs: BottomNavigationTab[]) {
    if (!this.tabs) { this.tabs = tabs; }
    const toUITabBarItem = (tab: BottomNavigationTab, tag: number) => UITabBarItem.alloc().initWithTitleImageTag(tab.title, fromResource(tab.icon).ios, tag);
    const bottomNavigationTabs: UITabBarItem[] = tabs.map(toUITabBarItem);

    this.nativeView.items = bottomNavigationTabs;
    this.nativeView.selectedItem = bottomNavigationTabs[this.selectedTabIndex];
  }



  [tabsProperty.setNative](value: BottomNavigationTab[]) {
    this.createTabs(value);
  }
```

**Comment by Pandishpan** [comment]
@henrychavez I created a pull request #66 with the changes that I think it should fix this problem.

**Comment by henrychavez** [comment]
I think you are calling the `createTabs()` function and also setting the tabs property in the component that's what is causing the issue, but I think the create tabs should clear the existing tabs and replaced with the last ones, and create another method like `addTab()` to add new tabs programmatically.

can you please send me your HTML to see if you are setting the tabs property there too?

**Comment by Pandishpan** [comment]
```
<GridLayout height="100%" rows="*, auto" width="100%">
    <page-router-outlet class="page page-content" left="0" tkMainContent top="0"></page-router-outlet>
    <BottomNavigation (tabSelected)="onBottomNavigationTabSelected($event)" [visibility]="menuVisible ? 'visible' : 'collapsed'" activeColor="#a1aa27" backgroundColor="white"
                     (loaded)="onBottomNavigationLoaded($event)" id="bottomNavigation" inactiveColor="black" keyLineColor="black" row="1" row="1" titleVisibility="selected">
    </BottomNavigation>
</GridLayout>

```

**Comment by Pandishpan** [comment]
>but I think the create tabs should clear the existing tabs and replaced with the last ones, and create another method like `addTab()` to add new tabs programmatically.

By the way, for the Android implementation, I think it needs some improvements on [bottom-navigation.android.ts](https://github.com/henrychavez/nativescript-bottom-navigation/blob/master/src/bottom-navigation.android.ts)
AHBottomNavigation has a method addItems which you are not using it. Because you are using the addItem for each tab there are a lot of instructions which are executed for each tab instead executing only once after all the tabs were added.
addItem method from AHBottomNavigation executes createItems()  after each call.
Please correct me if I'm wrong.

**Comment by Pandishpan** [comment]
To conclude this issue:

- On the Android platform, it's easily detectable because you are inserting tabs into the menu
- On iOS, you are replacing the array of items, so it's harder to notice the issue (but I'm sure that you are replacing them twice for no reason)

The route of the cause: when you set the tabs 
`if (!this.tabs) { this.tabs = tabs; }` the createTabs method it's called one more time.

> but I think the create tabs should clear the existing tabs and replaced with the last ones, and create another method like addTab() to add new tabs programmatically.

As you said, it would be nice to be able to replace all the tabs when the createTabs method is called. It's useful when you want to change your menu after you already created it.


Do you accept PRs?

**Comment by henrychavez** [comment]
Hi @Pandishpan ,

Yes I accept PRs but right now I'm working on a major update to remove `AHBottomNavigation` and user `BottomNavigationView` with support for Androidx  but I'll take your comments about this issue and included in the next release, that would be ready this weekend, also the reason for doing this is because the plugin now will be part of [nativescript-material components](https://github.com/Akylas/nativescript-material-components) that would be included in the official docs of Nativescript that were mentioned today in the launch of NS6. 😄

**Comment by henrychavez** [comment]
Fixed in version 2.0.0
